examples/experimental/streams.js

```diff
@@ -0,0 +1,34 @@
+import { ReadableStream } from 'k6/experimental/streams'
+import { setTimeout } from 'k6/timers'
+
+function numbersStream() {
+	let currentNumber = 0
+
+	return new ReadableStream({
+		start(controller) {
+			const fn = () => {
+				if (currentNumber < 5) {
+					controller.enqueue(++currentNumber)
+					setTimeout(fn, 1000)
+					return;
+				}
+
+				controller.close()
+			}
+			setTimeout(fn, 1000)
+		},
+	})
+}
+
+export default async function () {
+	const stream = numbersStream()
+	const reader = stream.getReader()
+
+	while (true) {
+		const { done, value } = await reader.read()
+		if (done) break
+		console.log(`received number ${value} from stream`)
+	}
+
+	console.log('we are done')
```

Comment:
Well.. yeah.. in practice, but while debugging WPTs is extremely useful, not to say that I'd even consider it a requirement, as it's the only way to debug the JS side (I don't think it's fair to consider goja debugger a real solution here, afaik).

So, do you really want me (or the person who needs to maintain that code) to copy and paste `console` code back & forth every time is needed? In the end, it's tests code...

Reply:
I usually create a helper such as `print` which can even call `spew.Dump` or just `fmt.Println` - which also lets you do custom stuff in case where you need it.

Even in some tests where we've needed it for the test itself ot function it makes it easier to have `print` that appends to a slice ... instead of having to do all the logrus plumbing to get the logs in the end.

Doing the plumbing makes sense when you are testing the whole of k6 running something ... but in htat case you  also already have the logger.

As I said - I am not against console being moved so that it can be imported by both, but I would argue this now is a liability someone will make tests using console.log and think it is the real thing and then it will have different behaviour in a real execution.

Reply:
I concur with @joanlopez on this one. I'd be interested in understanding if you see a concrete risk or downsides to adding out-of-the-box support for `console.log` in the test runtime.

Quite often, when implementing modules and writing tests for them, I need to be able to log values as I debug. The stream's implementation was just yet another occurrence.

While it's fair to argue that one could do some plumbing themselves every time they have this need, it doesn't feel like a good developer experience (for myself 😇 at the very least) to have to stop what I'm doing and switch to figuring out how to do it every single time. As a user, my expectation is: "My JS code is failing, and I need to see the value that's failing; my initial instinct is to resort to `console.log`." 
I've been using `throw` to do that, but in the streams API, where everything happens concurrently, it didn't scale that much. If I have to write a helper to do what I expect every time, I'd much rather have the tool I'd expect to find already present in the runtime to remove that overhead for myself.

This is a pretty common need when writing modules. Unless there's a downside to it, I think it would be a good quality-of-life improvement to have it as a default in the test runtime or at least to have a dedicated "rich" test runtime (side-by-side with the current one) or whatever that contains additional tooling such as this one.

Reply:
>  be interested in understanding if you see a concrete risk or downsides to adding out-of-the-box support for console.log in the test runtime.

>  As I said - I am not against console being moved so that it can be imported by both, but I would argue this now is a liability someone will make tests using console.log and think it is the real thing and then it will have different behaviour in a real execution.


So again - I am not against it. I am against it being reimplemented, and given that this PR does *not* use this - to be done in *this* PR.

The remaining post is reaffirming that *I* do not find it to be useful most of the time and showcases what I do. 

So if you want this - move this in another PR. But again given that this is the first time someone wanted it I would expect it won't be used a lot.

Reply:
I will resolve this - I guess we will be making an issue and/or a PR at some future point